python/paddle/fluid/executor.py

```diff
@@ -128,6 +128,94 @@ def as_numpy(tensor):
         return None
 
 
+def dtype_is_compatible_with(first, second):
+    """
+    Returns True if the first dtype can be compatible the second one.
+    Currently, we require the two dtype's have to be same.
+      
+    Args:
+        dtype (np.dtype|VarType|str): The type of data : float32, int64, etc.
+    
+    Returns:
+        True if the two types are same.
+    """
+    if not isinstance(first, core.VarDesc.VarType):
+        first = convert_np_dtype_to_dtype_(first)
+    if not isinstance(second, core.VarDesc.VarType):
+        second = convert_np_dtype_to_dtype_(second)
+    return first == second
+
+
+def dimension_is_compatible_with(first, second):
+    """
+    Returns True if the two dimensions are compatible.
+
+    A dimension is compatible with the other if:
+    1. The length of the dimensions are same.
+    2. Each non-negative number of the two dimentions are same.
+    3. For negative number or 'None' in a dimention, it means unknown so it
+       is compatible with any number.
+
+    Args:
+        first (list/tuple): integers representing shape. "None" or negative
+            number means unknown.
+        second (list/tuple): integers representing shape. "None" or negative
+            number means unknown.
+
+    Returns:
+        True if the two dimensions are compatible.
+    """
+
+    dim_len = len(first)
+    if dim_len != len(second):
+        return False
+
+    for i in range(dim_len):
+        if first[i] is None or first[i] < 0:
+            continue
+        if second[i] is None or second[i] < 0:
+            continue
+        if first[i] != second[i]:
+            return False
+
+    return True
+
+
+def check_feed_shape_type(var, feed):
+    """
+    Returns True if the variable doesn't require feed check or it is compatible
+    with the shape and have same dtype as the feeded value.
+
+    A dimension is compatible with the other if:
+    1. The length of the dimensions are same.
+    2. Each non-negative number of the two dimentions are same.
+    3. For negative number or 'None' in a dimention, it means unknown so it
+       is compatible with any number.
+    
+    Args:
+        var (Variable): the Variable object
+        feed (list|np.array): the feeded value
+    Returns:
+        True if the shape and dtype of variable is compatible with the feed value
+    Raises:
+        ValueError: if the shape or dtype of the variable is not compatible with
+            the feed value
+    """
+    if var.desc.need_check_feed():
+        numpy_feed = as_numpy(feed) if isinstance(
```

Comment:
`as_numpy` would need to copy data of feeded variables. It would be slow especially when the feeded data is in GPU.

Reply:
I found the feed instance won't be a LoDTensorArray, so I changed it. Now I only call numpy.asarray when input is list.